src/express-tooling.ts

```diff
@@ -112,25 +118,25 @@ const requireScopesMiddleware: requireScopesMiddleware =
           precedenceOptions.precedenceFunction :
           () => Promise.resolve(false);
 
-      precedenceFunction(request, response, nextFunction)
-      .catch(error => {
-        logOrNothing.warn('Error while executing precedenceFunction', error);
-        // PrecedencFunction was not successful
-        //  false -> trigger fallback to default scope validation
-        return false;
-      })
-      .then(isAllowed => {
-        if (isAllowed) { // precedenceFunction grants access
-          logOrNothing.debug('PrecedenceFunction grants access');
-          nextFunction();
-        } else if (validateScopes(request, scopes)) {  // scope validation grants access
-          logOrNothing.debug('Scopes validated successfully');
-          nextFunction();
-        } else {
-          logOrNothing.warn(`Scope validation failed for ${scopes}`);
-          authorizationFailedHandler(request, response, nextFunction, scopes, logOrNothing);
-        }
-      });
+      return precedenceFunction(request, response, nextFunction)
+        .catch(error => {
+          logOrNothing.warn('Error while executing precedenceFunction', error);
+          // PrecedencFunction was not successful
+          //  false -> trigger fallback to default scope validation
+          return false;
+        })
+        .then(isAllowed => {
+          if (isAllowed) { // precedenceFunction grants access
+            logOrNothing.debug('PrecedenceFunction grants access');
+            nextFunction();
+          } else if (validateScopes(request, scopes)) {  // scope validation grants access
+            logOrNothing.debug('Scopes validated successfully');
+            nextFunction();
+          } else {
+            logOrNothing.warn(`Scope validation failed for ${scopes.join(' ')}`);
```

Comment:
I suggest `, ` as separator, it's closer to the standard array formatter i.e. `[1, 2, 4]`